internal/test/e2e/version_compatibility_test.go

```diff
@@ -169,6 +170,102 @@ func TestVersionCompatibility(t *testing.T) {
 	})
 }
 
+func TestVersionCompatibility_FieldUpdatesEnsurePlugins(t *testing.T) {
+	cleanup := run.Koko(t)
+	defer cleanup()
+
+	dpCleanup := run.KongDP(kong.GetKongConfForShared())
+	defer dpCleanup()
+	require.NoError(t, util.WaitForKong(t))
+	require.NoError(t, util.WaitForKongAdminAPI(t))
+
+	kongAdmin, err := kongClient.NewClient(util.BasedKongAdminAPIAddr, nil)
+	require.NoError(t, err, "create go client for kong")
+	ctx := context.Background()
+	info, err := kongAdmin.Root(ctx)
+	require.NoError(t, err, "fetching Kong Gateway info")
+
+	dataPlaneVersion, err := versioning.NewVersion(kongClient.VersionFromInfo(info))
+	require.NoError(t, err)
+
+	admin := httpexpect.New(t, "http://localhost:3000")
+
+	// Remove plugins that may not be expected due to data plane version or aren't subject to compatibility updates
+	expectedPluginsMap := make(map[string]VersionCompatibilityPlugins, 0)
+	for _, pluginTest := range VersionCompatibilityOSSPluginConfigurationTests {
```

Comment:
I think this can be rewritten in a more efficient way:

```
    for _, pluginTest := range VersionCompatibilityOSSPluginConfigurationTests {
		if pluginTest.FieldUpdateChecks == nil {
			continue
		}
		if len(pluginTest.VersionRange) > 0 {
			version := versioning.MustNewRange(pluginTest.VersionRange)
			if !version(dataPlaneVersion) {
				continue
			}
		}
		expectedPluginsMap[pluginTest.Name] = pluginTest
	}
```

This way you don't need the extra `addPlugin` and avoid cycles when not needed